bitcoin/src/address/mod.rs

```diff
@@ -384,8 +384,11 @@ impl Address {
         Ok(Address::p2sh_from_hash(hash, network))
     }
 
-    // This is intentionally not public so we enforce script length checks.
-    fn p2sh_from_hash(hash: ScriptHash, network: impl Into<NetworkKind>) -> Address {
+    /// Creates a pay to script hash P2SH address from a script hash.
+    ///
+    /// **Warning** The `hash` pre-image (redeem script) must not exceed 520 bytes in length
```

Comment:
I think we should use the standard proposed in #2798:

```suggestion
    /// # Warning
    ///
    /// The `hash` pre-image (redeem script) must not exceed 520 bytes in length
```

Reply:
Oooo, good point. Thanks man, will do.

Reply:
Maybe plural to fit in with `Examples` and `Returns`?